## **Mô tả**
Bộ dữ liệu FoodPrice_in_Turkey chứa dữ liệu chứa dữ liệu về giá cả các mặt hàng thực phẩm tại Turkey, các thông tin được thống kê bao gồm:

Place	ProductId	ProductName	UmId	UmName	Month	Year	Price
+ Place: Nơi bán
+ ProductID: Mã sản phẩm
+ ProductName: Tên sản phẩm
+ UmId: Mã đơn vị đo lường
+ UmName: Tên đơn vị đo lường
+ Month: Tháng
+ Year: Năm
+ Price: Giá

Hãy đưa bộ dữ liệu vào phân tích và giải đáp các thắc mắc sau:

+ Bộ dữ liệu chứa bao nhiêu dòng, bao nhiêu cột
+ Thang đo tương ứng của các thuộc tính được lưu trữ
+ Giá trung bình của từng loại thực phẩm là bao nhiêu
+ Đề xuất những loại phân tích mà bạn muốn biết mà bộ dữ liệu này có thể cung cấp

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
fp = pd.read_csv('FoodPrice_in_Turkey.csv')

In [3]:
fp.head()

Place  ProductId    ProductName  UmId UmName  Month  Year  \
0  National Average         52  Rice - Retail     5     KG      5  2013   
1  National Average         52  Rice - Retail     5     KG      6  2013   
2  National Average         52  Rice - Retail     5     KG     11  2013   
3  National Average         52  Rice - Retail     5     KG     12  2013   
4  National Average         52  Rice - Retail     5     KG      1  2014   

    Price  
0  4.4920  
1  4.5786  
2  4.7865  
3  5.1337  
4  5.5099

In [4]:
fp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7381 entries, 0 to 7380
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Place        7381 non-null   object 
 1   ProductId    7381 non-null   int64  
 2   ProductName  7381 non-null   object 
 3   UmId         7381 non-null   int64  
 4   UmName       7381 non-null   object 
 5   Month        7381 non-null   int64  
 6   Year         7381 non-null   int64  
 7   Price        7381 non-null   float64
dtypes: float64(1), int64(4), object(3)
memory usage: 461.4+ KB


- Bộ dữ liệu chứa 7381 dòng, 8 cột

Nhận diện thang đo cho từng biến quan sát

- Place, ProductName, UmName: có kiểu object –> norminal attribute
- Month, Year: là dữ liệu dạng số nguyên (int64) mô tả tháng, năm. Quan sát, những thuộc tính này chỉ dùng để định danh, phân biệt giữa tháng, năm khác nhau –> norminal attribute
- ProductId, UmId mặc dù có kiểu số (int64), nhưng các con số này chỉ có ý nghĩa phân biệt khách hàng này với khách hàng khác –> thuộc tính định danh
- Price: thuộc tính định lượng, thang đo ratio

### **Chỉnh sửa cấu trúc bộ dữ liệu**

In [8]:
# Đổi tên cột thuộc tính, inplace = True đổi tiếp trên dữ liệu gốc mà không phải tạo bản sao
fp.rename(columns={"Place": "Địa điểm", "ProductName": "Tên SP"},inplace=True)
fp.head()

Địa điểm  ProductId         Tên SP  UmId UmName  Month  Year  \
0  National Average         52  Rice - Retail     5     KG      5  2013   
1  National Average         52  Rice - Retail     5     KG      6  2013   
2  National Average         52  Rice - Retail     5     KG     11  2013   
3  National Average         52  Rice - Retail     5     KG     12  2013   
4  National Average         52  Rice - Retail     5     KG      1  2014   

    Price Giảm giá  
0  4.4920      10%  
1  4.5786      10%  
2  4.7865      10%  
3  5.1337      10%  
4  5.5099      10%

In [9]:
# Thêm cột giảm giá 10% cho tất cả các bản ghi
fp['Giảm giá'] = '10%'
fp.head()

Địa điểm  ProductId         Tên SP  UmId UmName  Month  Year  \
0  National Average         52  Rice - Retail     5     KG      5  2013   
1  National Average         52  Rice - Retail     5     KG      6  2013   
2  National Average         52  Rice - Retail     5     KG     11  2013   
3  National Average         52  Rice - Retail     5     KG     12  2013   
4  National Average         52  Rice - Retail     5     KG      1  2014   

    Price Giảm giá  
0  4.4920      10%  
1  4.5786      10%  
2  4.7865      10%  
3  5.1337      10%  
4  5.5099      10%

In [10]:
fp['Giảm giá'].dtypes

dtype('O')

In [ ]:
pd.DataFrame

### **Giá trung bình của từng loại thực phẩm là bao nhiêu**

In [ ]:
# Hint: sử dụng groupby và mean để tính giá trung bình của từng loại thực phẩm
mean_productname = fp.groupby('ProductName').mean().sort_values(['Price', 'UmId'], ascending = False)
mean_productname[['Price', 'UmId']].head(10)

In [ ]:
fp.ty